sdk/trace/src/test/java/io/opentelemetry/sdk/trace/SdkSpanBuilderTest.java

```diff
@@ -944,8 +944,8 @@ void spanDataToString() {
     assertThat(span.toSpanData().toString())
         .matches(
             "SpanData\\{spanContext=ImmutableSpanContext\\{"
-                + "traceId=[0-9a-f]{32}, "
-                + "spanId=[0-9a-f]{16}, "
+                + "traceId=(?!00000000000000000000000000000000)[0-9a-f]{32}, " // check not invalid
```

Comment:
Is toString affected by this change?

Reply:
the negative lookahead is just to make sure it's not an invalid trace id, which would have caught my very bad boolean logic (I should have expanded on that comment to the right, but ran out of room 😂)

Reply:
so... you got the boolean logic wrong, but I should trust you to get the regex right?  🤣

Reply:
I kid, I kid. :)

Reply:
I promise 😂, I updated the test, and the test failed, then I fixed the boolean logic, and the test passed

Reply:
Ah ok - the invalid spancontext is usually replaced by a valid one here I'm presuming which stopped happening with the wrong logic.

It's probably worth having a unit test that calls the `create` method directly setting the bool true or false